pandas/core/arrays/categorical.py

```diff
@@ -1686,13 +1687,20 @@ def __array__(
         >>> np.asarray(cat)
         array(['a', 'b'], dtype=object)
         """
+        if copy is False:
+            raise ValueError(
+                "Unable to avoid copy while creating an array as requested."
+            )
+
         ret = take_nd(self.categories._values, self._codes)
-        if dtype and np.dtype(dtype) != self.categories.dtype:
-            return np.asarray(ret, dtype)
```

Comment:
I did not understand why this is needed.  If dtypes match, NumPy should make it a no-op?  If dtype is None, it is the same as not passing.

Reply:
Yeah, also not sure why this is here